keras/utils/vis_utils.py

```diff
@@ -283,9 +283,8 @@ def format_shape(shape):
             [format_shape(ishape) for ishape in layer.input_shapes])
       else:
         inputlabels = '?'
-      label = '%s\n|{input:|output:}|{{%s}|{%s}}' % (label,
-                                                     inputlabels,
-                                                     outputlabels)
+      label = f'{label}|{{input:|output:}}|' \
+          + f'{{{{{inputlabels}}}|{{{outputlabels}}}}}'
```

Comment:
Agreed that it does not look very readable. Please go with the option you think works best.

To put a string across two lines, prefer doing

```
st = ('first part'
         'second part')
```

Reply:
Updated, thanks.